tests/console/yast2_registration.pm

```diff
@@ -0,0 +1,64 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: check if an unregistered system can be registered and if
+#          enabling and disabling extensions correctly work.
+# Maintainer: Paolo Stivanin <pstivanin@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+
+sub register_system {
+    wait_screen_change { type_string get_var "SCC_EMAIL" };
+    send_key "tab";
+    wait_screen_change { type_string get_var "SCC_REGCODE" };
+    send_key "alt-n";
+    assert_screen("yast2_registration-ext-mod-selection", timeout => 60);
+    # enable Web and Scripting Module
+    for my $i (0 .. 25) {
+        send_key "down";
+    }
+    send_key "spc";
+    send_key "alt-n";
+    wait_still_screen(stilltime => 7, timeout => 60);
+    if (check_screen "yast2_registration-license-agreement") {
+        send_key "alt-a";
+        send_key "alt-n";
+        wait_still_screen 5;
+    }
+    send_key "alt-a";
+    wait_still_screen 2;
+    send_key "alt-o";
+    wait_still_screen 5;
+    wait_screen_change { send_key "alt-f" };
+}
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    zypper_call "in yast2-registration";
+
+    script_run("SUSEConnect -d",        15);
```

Comment:
Why 15 ? it is failing in your last verification run ?

Reply:
* I had to cleanup the openqa instance in order to free some space, that's why a lot of stuff got deleted
* you can't see the code because I'm actually testing other stuff, so the branch containing this code is no longer there (but you can see it here https://github.com/paolostivanin/os-autoinst-distri-opensuse/blob/y2reg-poo48569/tests/console/yast2_registration.pm)
* the `15` is probably there since I started developing the test (don't remember why I put it there TBH). I'll remove it :+1:

Reply:
it's bad when you have to cleanup your instance so often
ok tests passed, but there were obviously failures with `script_run` which do not break whole test
when it is supposed to pass, I would just use `assert_script_run`

Reply:
problem is that `SUSEConnect -d` may fail, that's why I decided to use script_run instead.
I'll run a new set of tests once I'm done with the other test I'm working on :)

Reply:
do you need to run `SUSEConnect -d` then when you run `SUSEConnect --cleanup` ?

Reply:
according to the help page, yes. The former is for deregistering the system, while the latter is for cleaning up creds and services

Reply:
Well if you would be customer i would agree, but this is test system and you don't care that much about de-registration. I think just `--cleanup` is painless same as here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/registration.pm#L178

Reply:
Good point :+1: